pysat/utils/io.py

```diff
@@ -172,121 +167,555 @@ def filter_netcdf4_metadata(inst, mdata_dict, coltype, remove=False,
-                # This is a Series or DataFrame, possibly with more dimensions.
-                # Series and DataFrame data must be treated differently.
-                try:
-                    # Assume it is a DataFrame and get a list of subvariables
-                    subvars = inst[var].iloc[0].columns
-                    is_frame = True
-                except AttributeError:
-                    # Data is Series of Series, which doesn't have columns
-                    subvars = [inst[var].iloc[0].name]
-                    is_frame = False
-
-                # Get the dimensions and their names
-                dims = np.shape(inst[var].iloc[0])
-                obj_dim_names = []
-                if len(dims) == 1:
-                    # Pad the dimensions so that the rest of the code works
-                    # for either a Series or a DataFrame
-                    dims = (dims[0], 0)
-
-                for i, dim in enumerate(dims[:-1]):
-                    # Don't need to go over last dimension value,
-                    # it covers number of columns (if a DataFrame)
-                    obj_dim_names.append(var)
-
-                # Set the base-level meta data
-                meta_dict['Depend_1'] = obj_dim_names[-1]
-
-                # Cycle through each of the sub-variable, updating metadata
-                for svar in subvars:
-                    # Get the correct location of the sub-variable based on
-                    # the object type
-                    if is_frame:
-                        idx = inst[var].iloc[0][svar]
-                    else:
-                        idx = inst[var].iloc[0]
+    """
 
-                    # Attach the metadata
-                    _, scoltype, _ = inst._get_data_info(idx)
+    # Define a default translation
+    trans_table = {}
 
-                    smeta_dict = {'Depend_0': epoch_name,
-                                  'Depend_1': obj_dim_names[-1],
-                                  'Display_Type': 'Spectrogram',
-                                  'Format': inst._get_var_type_code(scoltype),
-                                  'Var_Type': 'data'}
-                    inst.meta[svar] = smeta_dict
+    # Update labels required by netCDF4
+    trans_table['_FillValue'] = meta.labels.fill_val
+    trans_table['FillVal'] = meta.labels.fill_val
+
+    # We *may* need to keep this for backwards compatibility. Unintended use
+    # of 'fill' in pysat generated files.
+    trans_table['fill'] = meta.labels.fill_val
+
+    return trans_table
+
+
+def default_to_netcdf_translation_table(inst):
+    """Return metadata translation table with minimal netcdf requirements.
+
+    Parameters
+    ----------
+    inst : pysat.Instrument
+        Instrument object to be written to file.
+
+    Returns
+    -------
+    dict
+        Keyed by self.labels with a list of strings to be used
+        when writing netcdf files.
+
+    """
+
+    # Define a default translation
```

Comment:
Init dict with line 521 or something like `trans_table = {val: [val] for val in inst.meta.labels.label_attrs.keys()}`